clients/fides-js/src/components/tcf/TcfVendors.tsx

```diff
@@ -157,35 +162,39 @@ const StorageDisclosure = ({ vendor }: { vendor: Vendor }) => {
 };
 
 const TcfVendors = ({
-  allVendors,
-  allSystems,
-  enabledVendorIds,
-  enabledSystemIds,
+  vendors,
+  enabledVendorConsentIds,
+  enabledVendorLegintIds,
   onChange,
   gvl,
 }: {
-  allVendors: PrivacyExperience["tcf_vendors"];
-  allSystems: PrivacyExperience["tcf_systems"];
-  enabledVendorIds: string[];
-  enabledSystemIds: string[];
+  vendors: PrivacyExperience["tcf_vendors"];
+  enabledVendorConsentIds: string[];
+  enabledVendorLegintIds: string[];
   onChange: (payload: UpdateEnabledIds) => void;
   gvl?: GVLJson;
 }) => {
   const [isFiltered, setIsFiltered] = useState(false);
 
-  // Vendors and Systems are the same for the FE, but are 2 separate
-  // objects in the backend. We combine them here but keep them separate
-  // when patching preferences
-  const vendors = [...(allVendors || []), ...(allSystems || [])];
-  const enabledIds = [...enabledVendorIds, ...enabledSystemIds];
-
-  if (vendors.length === 0) {
```

Comment:
I snuck in a refactor here because it was getting unwieldy to pass in `enabledVendorConsentIds`, `enabledVendorLegintIds`, `enabledSystemLegintIds`, `enabledSystemConsentIds`. instead, the parent component combines vendors and systems (this component used to do that). the parent component is also responsible for separating back out into vendors and systems to save to the backend.